mautic/mautic
# [DPMMA-2187] Segment date filter and "equals" operator

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [✅]
| New feature/enhancement? (use the a.x branch)      | [❌]
| Deprecations?                          | [❌]
| BC breaks? (use the c.x branch)        | [❌]
| Automated tests included?              | [✅] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

This PR extends the validation of data fields using the equals/not equal operators. Using these operators for dates causes the segment to not rebuild properly. This is due to the fact that date-related data is stored in the form of datetime and comparing it with only the date or incomplete datetime in the query does not return the expected results. The value to be compared must be accurate to the second, comparison with only the day in the YYYY-MM-DD format or with the day and partial time does not return all relevant contacts. This PR does not remove operators for data fields so that segments that have already been saved with them can be opened for editing.



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create new segment or edit existing one.
3. Add a filter related to the date, e.g. Date added, select the "equals" or "not equal" operator and fill in the sample value  "2024-04-15"
4. Try saving the segment. The following message should appear
![date_added](https://github.com/mautic/mautic/assets/102536220/7c89deef-e0cd-4a3b-b53b-5f769927255b)
